ai_genomics/analysis/influence/influence_analysis.py

```diff
@@ -0,0 +1,219 @@
+        lambda corpus: [tm.remove_short_tokens(d) for d in corpus],
+        partial(tm.build_ngrams, n=n_grams),
+        lambda tok_output: {
+            _id: tok
+            for _id, tok in zip(text_dict.keys(), tok_output[0])
+            if len(tok) > 0  # Remove empty documents
+        },
+    )
+
+
+def make_topic_distribution(
+    text_dict: Dict[str, str], num_topics: int = 100, top_remove: int = 50
+) -> pd.DataFrame:
+    """Fits topic model and creates topic distribution
+
+    Args:
+        text_dict: Dictionary of text to process.
+        num_topics: Number of topics to fit.
+        top_remove: Number of top words to remove from topic model.
+
+    Returns:
+        Topic distribution for the corpus
+    """
+    lda_model = tm.train_lda(list(text_dict.values()), k=num_topics, top_remove=50)
+
+    # Create topic dataframe
+    return tm.create_doc_topics(
+        lda_model,
+        tm.create_topic_names(lda_model, num_topics),
+        list(text_dict.keys()),
+    )
+
+
+def calculate_topic_shares(
+    topic_df: pd.DataFrame,
+    disc_names_ids: Dict[str, List],
+    baseline_discs: List[str] = ["ai", "genomics"],
+    binarise_thres: float = 0.9,
+) -> pd.DataFrame:
+    """Calculates the topic shares for each category
+
+    Args:
+        topic_df: Topic distribution for the corpus.
+        disc_names_ids: Dictionary of disc names and ids.
+
+    Returns:
+        Df of discipline shares in each topic
+    """
+    # Calculate the topic shares for each category
+
+    return (
+        topic_df.apply(lambda top: binarise_top(top, q=binarise_thres))
+        .assign(topic_category=lambda df: df.index.map(disc_names_ids))
+        .melt(id_vars="topic_category")
+        .groupby(["topic_category", "variable"])["value"]
+        .sum()
+        .unstack(level=0)[baseline_discs]
+        .apply(lambda x: x / x.sum(), axis=1)
+        .stack()
+        .reset_index(name="disc_weight")
+    )
+
+
+def calculate_influence_score(
+    topic_df: pd.DataFrame, disc_shares: pd.DataFrame
+) -> pd.DataFrame:
+    """Calculates the influence score for each document
+
+    Args:
+        topic_df: Topic distribution for the corpus.
+        disc_names_ids: Dictionary of disc names and ids.
+        disc_shares: Topic shares for each category.
+
+    Returns:
+        DF of influence score for each document
+    """
+    return (
+        topic_df.stack()
+        .reset_index(name="topic_weight")
+        .rename(columns={"level_1": "topic", "level_0": "doc_id"})
+        .merge(disc_shares, left_on="topic", right_on="variable")
+        .drop(axis=1, labels=["variable"])
+        .assign(topic_disc_weighted=lambda df: df["topic_weight"] * df["disc_weight"])
+        .groupby(["doc_id", "topic_category"])["topic_disc_weighted"]
+        .sum()
+    )
+
+
+if __name__ == "__main__":
+    logging.info("testing with the GtR data")
+
+    import ai_genomics.getters.gtr as gtr
+
+    gtr_meta = gtr.get_ai_genomics_project_table()
```

Comment:
just to note, this initially failed for me as the default in the getter is local. Not really for this PR but i suppose there's an assumption that whoever is runnning this has run all steps up to this point and so has all output files locally, that we should note/clean up further down the line